src/App.jsx

```diff
@@ -14,33 +15,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const prepareGoodsList = (goods, sortOrder, isReversed) => {
+  let sortedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortOrder === 'alphabetically') {
+    sortedGoods.sort();
+  } else if (sortOrder === 'length') {
+    sortedGoods.sort((a, b) => a.length - b.length);
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortOrder, setSortOrder] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goodsToRender = prepareGoodsList(goodsFromServer, sortOrder, isReversed);
+
+  const sortAlphabetically = () => {
+    setSortOrder('alphabetically');
+  };
+  const sortByLength = () => {
+    setSortOrder('length');
+  };
+
+  const reverseOrder = () => {
+    setIsReversed((prev) => !prev);
+  };
+
+  const resetOrder = () => {
+    setSortOrder('');
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortOrder === 'alphabetically' ? '' : 'is-light'}`}
```

Comment:
use classnames library in next tasks for dynamic class handling.
https://www.npmjs.com/package/classnames